app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -48,6 +48,33 @@ def perform_auto_provision host, rule
     end
   end
 
+  def perform_reboot_all hosts = Host::Discovered.all
+    result = true
+    error_message = _("Errors during reboot: %s")
+    overall_errors = ""
+
+    if hosts.count > 0
+      Host.transaction do
+        hosts.each do |discovered_host|
+          begin
+            unless discovered_host.reboot
+              overall_errors << "#{discovered_host.name}: failed to reboot "
```

Comment:
@lzap all the errors are returned to the controller so maybe there should be one error there instead of writing each error to the log separately?